### DISCUSSION TITLE
1.2.0

### DISCUSSION THREAD

**Discussion by stephencelis** [pointfreeco/swift-issue-reporting#89] [General]
## What's Changed

* Added: Issue Reporting (https://github.com/pointfreeco/swift-issue-reporting/pull/87).

    XCTest Dynamic Overlay is now Issue Reporting, a general purpose library for reporting issues in your application and library code as Xcode runtime warnings, breakpoints, assertions, and more, and do so in a testable manner, with support for both XCTest and Swift's new native Testing framework.

* Infrastructure: Swift Language Support: Drop <5.9, Add 6.0 by @stephencelis in https://github.com/pointfreeco/swift-issue-reporting/pull/84
* Infrastructure: Add `.editorconfig` for consistent code formatting by @Matejkob in https://github.com/pointfreeco/swift-issue-reporting/pull/85

## New Contributors
* @Matejkob made their first contribution in https://github.com/pointfreeco/swift-issue-reporting/pull/85

**Full Changelog**: https://github.com/pointfreeco/swift-issue-reporting/compare/1.1.2...1.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/pointfreeco/swift-issue-reporting/releases/tag/1.2.0'>1.2.0</a>.</em>

**Comment by bweissmann** [comment]
FYI some builds might be down during migration from xctest-dynamic-overlay to swift-issue-reporting

PSA in case other folks run into the same logs as us.

If anyone is seeing logs like this today: 
```
2024-07-22T23:39:08.8990600Z   multiple similar targets 'IssueReporting', 'IssueReportingTestSupport', 'XCTestDynamicOverlay' appear in package 'swift-issue-reporting' and 'xctest-dynamic-overlay', this may indicate that the two packages are the same and can be de-duplicated by using mirrors. if they are not duplicate consider using the `moduleAliases` parameter in manifest to provide unique names
```
You are likely depending on `swift-url-routing` [link](https://github.com/pointfreeco/swift-url-routing) and the migration from XCTestDynamicOverlay to SwiftIssueReporting broke builds until dependencies are updated like so:
https://github.com/pointfreeco/swift-url-routing/pull/104

**Reply by stephencelis** [reply]
@bweissmann We should have ushered out releases for all of our libraries, so a `swift package update` should hopefully address the above. Let us know if we're missing a library, though!

**Reply by niexyr** [reply]
This temporarily resolved our issue for us:

Step 1. Discard/Delete `Package.resolved` file.
Step 2. `Reset package caches`.

If you have `xctest-dynamic-overlay` this library as a transative dependency, this might work.

If `xctest-dynamic-overlay`'s version is 1.2.1/1.2.0 it will give you the error in the description above.
With this solution our version was 1.1.2 and everything works without any problem.

Hope this helps.

**Reply by ylorn** [reply]
@stephencelis we run into the exact same problem. Due to the company policy, we are now fixing `swift-composable-architecture` to `1.3.0` for it being the latest version before integrations of macros. We cannot upgrade `swift-composable-architecture` to the latest version until the security team green light the use of macros. Now our pipeline is blocked due to this change. Can you have a legacy repo with name `xctest-dynamic-overlay` for the transition?

**Reply by stephencelis** [reply]
@buscarini A thumbs down reaction isn't super helpful feedback if you want us to help troubleshoot a problem. Want to let us know what's wrong?

@ylorn SwiftPM supports redirects, and https://github.com/pointfreeco/xctest-dynamic-overlay redirects to swift-issue-reporting, so another repo shouldn't be necessary here. 1.2.0 had updated the package's name in Package.swift, which was the main issue affecting folks that were only able to partially update their dependencies. So we ushered out fixes in 1.2.1 and 1.2.2 that revert the package name in Package.swift to ensure better backwards compatibility when there are repo name mismatches in your dependency hierarchy. Can you try pointing to 1.2.2 and confirm if that helps get TCA 1.3.0 building for you again?

**Reply by ylorn** [reply]
@stephencelis Thanks, fixing `xctest-dynamic-overlay` version to 1.2.2 with `xcodebuild` flag `-disableAutomaticPackageResolution` unblocked our pipeline for now.

**Comment by stephencelis** [comment]
We are in the process of releasing patches that pin everything to the old repo name. While we did test many upgrade scenarios for a package rename before moving forward with this release, we are aware that it's causing unforeseen problems in certain setups, and so we think the only safe path forward is to leave xctest-dynamic-overlay as is and extract swift-issue-reporting to a new repository. These releases will complete rolling out shortly! We appreciate your patience.

**Reply by stephencelis** [reply]
The rollout has completed with https://github.com/pointfreeco/swift-composable-architecture/releases/tag/1.12.1

Please let us know if you hit any other issues!

**Comment by MeT3ger** [comment]
what is easy way to resolve this problem?

**Reply by stephencelis** [reply]
@MeT3ger There isn't a single problem here, so it depends on your scenario, but if you update your package dependencies ideally you will be in a building state. If not, please report more information back here.

**Reply by MeT3ger** [reply]
i'm use ubuntu and create my project on Vapor swift.  Today im update my packages and break my brains with this:
error: multiple similar targets 'IssueReporting', 'IssueReportingTestSupport', 'XCTestDynamicOverlay' appear in package 'swift-issue-reporting' and 'xctest-dynamic-overlay', this may indicate that the two packages are the same and can be de-duplicated by using mirrors. if they are not duplicate consider using the `moduleAliases` parameter in manifest to provide unique names.
how i can fix it? if u need another info, i can do it

**Reply by MeT3ger** [reply]
swift url routing - 0.6.1.  xctest-dynamic-overlay - 1.2.2 swift issue reporting - 1.2.2

**Reply by stephencelis** [reply]
@MeT3ger swift-url-routing 0.6.2 was just cut. Can you try again?

**Reply by MeT3ger** [reply]
ohh. I feel happy bunny. You are a magician for me! 😀😀

**Reply by MeT3ger** [reply]
the knives are sharpened, the smoke is smoking. Swift flies 😜
